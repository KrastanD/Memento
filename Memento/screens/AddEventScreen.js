import React, { useState } from "react";
import { Text, View, Alert } from "react-native";
import postEvent from "../server/postEventAxios";
import { TextInput } from "react-native-paper";
import DateTimePicker from "@react-native-community/datetimepicker";
import { TouchableOpacity } from "react-native-gesture-handler";
import Icon from "react-native-vector-icons/MaterialCommunityIcons";
import GetLocation from "react-native-get-location";

export default function AddEventScreen({ navigation }) {
  const [evTitle, setEvTitle] = useState("");
  const [evDesc, setEvDesc] = useState("");
  const [evLoc, setEvLoc] = useState("");
  const [date, setDate] = useState(Date.now());
  const [mode, setMode] = useState("date");
  const [show, setShow] = useState(false);

  const onChange = (event, selectedDate) => {
    const currentDate = selectedDate || date;
    setShow(Platform.OS === "ios");
    setDate(currentDate);
  };

  const showMode = (currentMode) => {
    setShow(true);
    setMode(currentMode);
  };

  const showDatepicker = () => {
    showMode("date");
  };

  const showTimepicker = () => {
    showMode("time");
  };

  let numberOfLines = 0;
  return (
    <View style={{ margin: 10 }}>
      <TextInput
        mode="outlined"
        theme={{
          colors: { primary: "#01a699" },
        }}
        label="Enter event title*"
        style={{
          height: 40,
        }}
        onChangeText={(text) => setEvTitle(text)}
        value={evTitle}
      />

      <TextInput
        label="Enter description"
        theme={{
          colors: { primary: "#01a699" },
        }}
        mode="outlined"
        multiline
        onContentSizeChange={(event) => {
          numberOfLines = event.nativeEvent.contentSize.height / 18;
        }}
        numberOfLines={numberOfLines}
        onChangeText={(text) => setEvDesc(text)}
        value={evDesc}
      />
      <View style={{ flexDirection: "row", alignItems: "center" }}>
        <TextInput
          label="Enter location"
          theme={{
            colors: { primary: "#01a699" },
          }}
          mode="outlined"
          style={{
            height: 40,
            flex: 1,
          }}
          onChangeText={(text) => setEvLoc(text)}
        />
        <TouchableOpacity
          style={{ paddingHorizontal: 5 }}
          onPress={() => {
            navigator.geolocation.getCurrentPosition(
              (position) => {
                setEvLoc(
                  position["coords"]["latitude"] +
                    " " +
                    position["coords"]["longitude"]
                );
              },
              (error) => Alert.alert(error.message),
              { enableHighAccuracy: true, timeout: 15000, maximumAge: 1000 }
            );
          }}
        >
          <Icon name="crosshairs-gps" size={40}></Icon>
        </TouchableOpacity>
      </View>
      <View style={{ paddingTop: 5 }}>
        <Text> Date and time: {getFormattedDate(date)}</Text>
        {evLoc ? <Text> {"Location: " + evLoc} </Text> : null}
      </View>
      <View style={{ paddingTop: 5 }}>
        <TouchableOpacity
          style={{
            height: 50,
            alignItems: "center",
            backgroundColor: "#01a699",
            elevation: 2,
          }}
          onPress={showDatepicker}
        >
          <Text
            style={{
              fontSize: 20,
              padding: 10,
              color: "white",
            }}
          >
            Change date
          </Text>
        </TouchableOpacity>
      </View>
      <View style={{ paddingTop: 5 }}>
        <TouchableOpacity
          onPress={showTimepicker}
          style={{
            height: 50,
            alignItems: "center",
            backgroundColor: "#01a699",
            elevation: 2,
          }}
        >
          <Text
            style={{
              fontSize: 20,
              padding: 10,
              color: "white",
            }}
          >
            Change time
          </Text>
        </TouchableOpacity>
      </View>

      {show && (
        <DateTimePicker
          maximumDate={Date.now()}
          testID="dateTimePicker"
          timeZoneOffsetInMinutes={0}
          value={date}
          mode={mode}
          is24Hour={true}
          display="default"
          onChange={onChange}
        />
      )}
      <View style={{ paddingTop: 15 }}>
        <TouchableOpacity
          style={{
            backgroundColor: "#01a699",
            height: 50,
            alignSelf: "center",
            backgroundColor: "#01a699",
            elevation: 2,
          }}
          onPress={() => {
            const d = new Date(date);
            postEvent(evTitle, evDesc, evLoc, d.toString());
            navigation.navigate("Events");
          }}
        >
          <Text
            style={{
              fontSize: 20,
              padding: 10,
              color: "white",
            }}
          >
            Submit
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

function getFormattedDate(date) {
  const d = new Date(date);
  const dtf = d.toDateString() + " " + d.toLocaleTimeString();
  return dtf;
}
