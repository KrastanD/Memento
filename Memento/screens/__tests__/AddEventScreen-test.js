import React from "react";
import AddEventScreen from "../AddEventScreen";
import renderer from "react-test-renderer";

beforeAll(() => {
  global.Date.now = jest.fn(() => new Date("2020-04-07T10:20:30Z").getTime());
});

it("snapshots AddEvent screen correctly", () => {
  const tree = renderer.create(<AddEventScreen />);
  expect(tree.toJSON()).toMatchSnapshot();
});

afterAll(() => {
  global.Date.now = Date.now();
});
