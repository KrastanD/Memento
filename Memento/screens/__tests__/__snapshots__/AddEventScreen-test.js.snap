// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots AddEvent screen correctly 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <View
    style={Object {}}
  >
    <View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            Object {
              "backgroundColor": "#f6f6f6",
              "borderColor": "rgba(0, 0, 0, 0.54)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "minHeight": 40,
            "paddingBottom": 0,
            "paddingTop": 8,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "bottom": 0,
                "height": 2,
                "left": 10,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 8,
              }
            }
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 29,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Enter event title*
          </Text>
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#01a699",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Enter event title*
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 1,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 28,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Enter event title*
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#01a699"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "textAlign": "left",
                "zIndex": 1,
              },
              Object {
                "height": 40,
              },
              Object {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlignVertical": "center",
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            Object {
              "backgroundColor": "#f6f6f6",
              "borderColor": "rgba(0, 0, 0, 0.54)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "minHeight": 64,
            "paddingBottom": 0,
            "paddingTop": 8,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "bottom": 0,
                "height": 2,
                "left": 10,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 8,
              }
            }
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 37,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Enter description
          </Text>
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#01a699",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Enter description
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 1,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Enter description
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={true}
          numberOfLines={0}
          onBlur={[Function]}
          onChangeText={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#01a699"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "textAlign": "left",
                "zIndex": 1,
              },
              Object {},
              Object {
                "paddingBottom": 22,
                "paddingTop": 22,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlignVertical": "top",
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 40,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 29,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Enter location
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#01a699",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 28,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Enter location
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 1,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 28,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Enter location
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#01a699"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 40,
                },
                Object {
                  "paddingBottom": 12,
                  "paddingTop": 12,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
            "paddingHorizontal": 5,
          }
        }
      >
        <Text />
      </View>
    </RNGestureHandlerButton>
  </View>
  <View
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    <Text>
       Date and time: 
      Tue Apr 07 2020 5:20:30 AM
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#01a699",
            "elevation": 2,
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "padding": 10,
            }
          }
        >
          Change date
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
  <View
    style={
      Object {
        "paddingTop": 5,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#01a699",
            "elevation": 2,
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "padding": 10,
            }
          }
        >
          Change time
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
  <View
    style={
      Object {
        "paddingTop": 15,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#01a699",
            "elevation": 2,
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "padding": 10,
            }
          }
        >
          Submit
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
